lib/ansible/plugins/lookup/oc.py

```diff
@@ -0,0 +1,252 @@
+      host:
+        description:
+          - The IP address of the host serving the OpenShift API
+        required: False
+        default: 127.0.0.1
+      port:
+        description:
+          - The port on which to access the OpenShift API
+        required: False
+        default: 8443
+      token:
+        description:
+          - The token to use for authentication against the OpenShift API.
+          - This can be a user or ServiceAccount token.
+        required: True
+      validate_certs:
+        description:
+          - Whether or not to validate the TLS certificate of the API.
+        required: False
+        default: True
+      namespace:
+        description:
+          - The namespace/project where the object resides.
+        required: False
+      resource_name:
+        description:
+          - The name of the object to query.
+        required: False
+      pretty:
+        description:
+          - Whether or not to prettify the output.  This is useful for
+          - debugging.
+        required: False
+        default: False
+      labelSelector:
+        description:
+          - Additional labels to include in the query.
+        required: False
+      fieldSelector:
+        description:
+          - Specific fields on which to query.
+        required: False
+      resourceVersion:
+        description:
+          - Query for a specific resource version.
+        required: False
+"""
+
+EXAMPLES = """
+- name: Get Project {{ project_name }}
+  set_fact:
+    project_fact:  "{{ lookup('oc',
+                       'Project',
+                       host=inventory_host,
+                       token=hostvars[inventory_host]['ansible_sa_token'],
+                       resource_name=project_name,
+                       validate_certs=validate_certs) }}"
+- name: Get All Service Accounts in a Project
+  set_fact:
+    service_fact: "{{ lookup('oc',
+                      'ServiceAccount',
+                      host=inventory_host,
+                      token=hostvars[inventory_host]['ansible_sa_token'],
+                      namespace=project_name,
+                      validate_certs=validate_certs) }}"
+"""
+
+RETURN = """
+  _list:
+    description:
+      - An object definition or list of objects definitions returned from OpenShift.
+    type: dict
+"""
+
+import json
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils import urls
+from ansible.module_utils.six.moves import urllib
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+
+class OC(object):
+    def __init__(self, host, port, token, validate_certs):
+        self.apis = ['api', 'oapi']
+        self.token = token
+        self.validate_certs = validate_certs
+        self.host = host
+        self.port = port
+        self.kinds = {}
+        bearer = "Bearer " + self.token
+        self.headers = {"Authorization": bearer}
+
+    def build_facts(self):
```

Comment:
Is there a reason this isn't a part of (could be called from) ```__init__()```?  It seems to only be used right after the class is instantiated.